dask/array/slicing.py

```diff
@@ -526,9 +557,18 @@ def take(outname, inname, blockdims, index, axis=0):
     {('y', 0): (getitem, ('x', 0), ([1, 3, 5],)),
      ('y', 2): (getitem, ('x', 2), ([7],))}
     """
-    if issorted(index):
```

Comment:
Yes, you are surely right. My idea was that if the index isn't sorted, we need to sort it anyway. So I was looking for a way to sort the index array that would also return a flag saying if the array was already sorted or not. I haven't found anything that does that though, so I will use your solution.